Jenkinsfile

```diff
@@ -177,30 +189,27 @@ pipeline {
         }
       }
       tools {
-        nodejs 'NodeJS 16'
+        nodejs 'NodeJS 20'
       }
       steps {
         script {
+          def version_before = npm.semver().version
           configFileProvider(NPMRC) {
             sh 'npm ci'
             sh 'npm run release'
           }
 
-          def tag = sh (
-            script: "./release-tool debug-RELEASE_VERSION",
-            returnStdout: true
-          ).split(' = ')[1].trim()
-
-          buildx.build(
-            project: PROJECT_NAME
-          , push: true
-          , tags: [tag]
-          , dockerfile: "distribution/docker/mezmo/Dockerfile"
-          )
+          def semver = npm.semver()
+          if (version_before != semver.version) {
+            buildx.build(
+              project: PROJECT_NAME
+            , push: true
+            , tags: [semver.version]
+            , dockerfile: "distribution/docker/mezmo/Dockerfile"
+            , docker_repo: DOCKER_REPO
+            )
+          }
```

Comment:
this step appears to be missing the bits to publish the MTP (what `./release-tool publish` was doing). is that something we're going to add back? or maybe I'm just missing it.

Reply:
Yep, that might have been missed.  Checking.

Reply:
This PR is actually 2 things:
1) use `buildx` and `docker.io` for images
2) Changes to support primary development off of this repo (ie use the resulting image in our SaaS)

I think we can go ahead and merge this to accomplish the first thing. It will fix a few nasty errors as it relates to keeping this repo in sync with the private one.  At a later time, there still has to be work to cut over the k8s manifest to using the new image, so at that time, we can add these missing MTP publishing bits.  Just don't wanna be blocked here now because of that.  Adding those steps might require tooling changes since everything has used `release-tool` in the recent past.  If that's the case, then we won't switch to this repo as our primary any time soon, but at least it'll be easy to merge with our private repo.